spine-c/src/spine/Skeleton.c

```diff
@@ -248,6 +261,54 @@ int spSkeleton_findBoneIndex (const spSkeleton* self, const char* boneName) {
 	return -1;
 }
 
+void _spSkeleton_addLastSlotToAnimations(spSkeleton* self)
+{
+	spAnimation *anim;
+	spTimeline *timeline;
+	spDrawOrderTimeline *drawOrderTL;
+	int	a, numAnims, t, numTL, f, numFrames;
+
+	numAnims = self->data->animationsCount;
+	for(a=0; a<numAnims; ++a)
+	{
+		anim = self->data->animations[a];
+
+		numTL = anim->timelinesCount;
+		for(t=0; t<numTL; ++t)
+		{
+			timeline = anim->timelines[t];
+			if(timeline->type == SP_TIMELINE_DRAWORDER)
+			{
+				drawOrderTL = SUB_CAST(spDrawOrderTimeline, timeline);
+				++CONST_CAST(int, drawOrderTL->slotsCount);
+
+				numFrames = drawOrderTL->framesCount;
+				for(f=0; f<numFrames; ++f)
+				{
+					drawOrderTL->drawOrders[f] = (int *) realloc(CONST_CAST(int*, drawOrderTL->drawOrders[f]), sizeof(int) * self->slotsCount);
+					CONST_CAST(int*, drawOrderTL->drawOrders[f])[self->slotsCount-1] = self->slotsCount-1;
+				}
+			}
+		}
+	}
+}
```

Comment:
The add functions aren't really needed for other runtimes, since other languages expose a list that can be mutated. Fixing up draw order timelines is different though. To merge this we'd need the functionality in all runtimes.